crates/nu-path/src/tilde.rs

```diff
@@ -152,7 +152,6 @@ fn expand_tilde_with_another_user_home(path: &Path) -> PathBuf {
 
 /// Expand tilde ("~") into a home directory if it is the first path component
 pub fn expand_tilde(path: impl AsRef<Path>) -> PathBuf {
-    // TODO: Extend this to work with "~user" style of home paths
```

Comment:
Is this working now? (I have no idea what `~user` style is)

Reply:
Not sure. Strangely enough, `ls ~` and `ls ~fdncred` return the same thing on my mac.

Reply:
oh my bad, I meant to comment on this in the PR description. yes, this is an old comment. this was implemented in #5251

Reply:
> I have no idea what `~user` style is

`~user` is expanded to the home directory of `user`.

> Strangely enough, `ls ~` and `ls ~fdncred` return the same thing on my mac.

That's how it works. Effectively, `~` is a shortcut to `~($env.USER)` where `$env.USER` is the current user.